### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by YATHENDRA1995**
Completed Exercises - 3.1, 3.2, 3.3, 3.4

Please review and add your comments

### CODE REVIEW

File: Week-3/Exercise(3.1)/memoize.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Timer Function
+const timer = (fn) => {
+  console.time();
+  console.log(fn());
+  console.timeEnd();
+};
+
+// Main Function
```
Comment (chirag-udeshi) [inline Week-3/Exercise(3.1)/memoize.js:8]: The below add function can only accepts 2 variables hence we can only add 2 numbers. But as per Guideline #4 -> The "function should be able to take n number of argument" .

File: Week-3/Exercise(3.4)/stack.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+function createStack() {
```
Comment (chirag-udeshi) [inline Week-3/Exercise(3.4)/stack.js:1]: The solution given is simple yet efficient. Also try implementing the solution using closures.

**Review by chirag-udeshi — APPROVED** [review]
Done great on all the 4 assignments. Additional review added on 2 exercise.

File: Week-3/Exercise(3.1)/memoize.js (hunk 1 above)
Comment (YATHENDRA1995) [reply Week-3/Exercise(3.1)/memoize.js:8]: Done - Implemented for n number of arguments

**Pull request merged**